addons/account/models/account_account.py

```diff
@@ -529,6 +529,7 @@ def _order_by_frequency_per_partner(self, company_id, partner_id, move_type=None
               WHERE account.company_id = %s
                    {where_internal_group}
             GROUP BY account.id
+              HAVING COUNT(aml.id) > 0
```

Comment:
make an inner join instead?

Reply:
Done :)
Furthermore, the "INNER JOIN" and "LIMIT 1"  are now conditional if `only_one` is set.
Otherwise, this would break https://github.com/odoo/odoo/blob/918871edf632a7a7ad20cc7a647ee689d338c480/addons/account/models/account_account.py#L544-L547
where no account suggestion at all would be displayed.